### PR TITLE
fix `clippy::uninlined_format_args`

### LINKED ISSUE #704: Fix or opt out of `clippy::uninlined_format_args`?
https://rust-lang.github.io/rust-clippy/master/#uninlined_format_args

Clippy now warns on this by default in Rust 1.66, while it was only added in [Rust 1.58](https://blog.rust-lang.org/2022/01/13/Rust-1.58.0.html) 48 weeks prior. I've been keeping it out of my own projects to not bump MSRV unnecessarily and to keep peers happy.

We'd be fine doing this though (in `ash/build.rs`!, the generator and examples) as our MSRV is already 1.59, but the question is whether we should. @Ralith how do you feel about the new format?

### PR DESCRIPTION
**Pull request by filnet**
Closes #704

Note that there is an unrelated `clippy::seek_to_start_instead_of_rewind` fix.

~And one instance of `clippy::uninlined_format_args` in generator.rs cannot be fixed and must be allowed.~

### CODE REVIEW

File: generator/src/lib.rs [hunk 1]
```diff
@@ -2379,6 +2379,11 @@ pub fn extract_native_types(registry: &vk_parse::Registry) -> (Vec<(String, Stri
         match ty.category.as_deref() {
             Some("include") => {
                 // `category="include"` lacking an `#include` directive are generally "irrelevant" system headers.
+                // using inlined format args in panic messages is not yet supported and will generate this warning:
+                // warning: panic message contains an unused formatting placeholder
+                // note: this message is not used as a format string when given without arguments, but will be in Rust 2021
+                // note: `#[warn(non_fmt_panics)]` on by default
+                #[allow(clippy::uninlined_format_args)]
```
Comment (Ralith) [inline generator/src/lib.rs:2386]: Could we fix this by updating the edition to 2021?

Comment (filnet) [reply generator/src/lib.rs]: Done.

File: generator/src/lib.rs [hunk 2]
```diff
@@ -1772,7 +1768,7 @@ pub fn derive_setters(
                         #[inline]
                         pub fn #param_ident_short(mut self, #param_ident_short: &'a #mutable #slice_param_ty_tokens) -> Self {
                             #set_size_stmt
-                            self.#param_ident = #param_ident_short#ptr;
+                            self.#param_ident = #param_ident_short #ptr;
```
Comment (filnet) [inline generator/src/lib.rs:1771]: See https://doc.rust-lang.org/edition-guide/rust-2021/reserving-syntax.html

**Review by Ralith — APPROVED** [review]
Thanks!

**Comment by filnet** [comment]
`clippy::uninlined_format_args` is going to be downgraded to pedantic : https://github.com/rust-lang/rust/blob/stable/RELEASES.md#version-1671-2023-02-09

**Review by MarijnS95 — APPROVED** [review]

**Pull request merged**